net/message/message.go

```diff
@@ -225,10 +225,30 @@ func magicVerify(magic uint32) bool {
 	return true
 }
 
-func PayloadLen(buf []byte) int {
+func PayloadLen(buf []byte) (int, []byte) {
 	var h msgHdr
 	h.Deserialization(buf)
-	return int(h.Length)
+	if magicVerify(h.Magic) {
+		//TODO: verify hdr checksum
+
+		return int(h.Length), buf
+	} else {
+		log.Fatal("========================================================================")
+		log.Fatal("Magic number wrong ,try to find the right index,now buf len is ", len(buf))
+		//try find the magic number
+		for i := 0; i <= len(buf)-MSGHDRLEN; i++ {
```

Comment:
If we just return error and open a new connection,  plenty of connection will close and reconnect continuously under a high throughput network, . Recover the stream would keep the connection state if the network back to normal, and the link would not lost if we just come across an accidental mistake buffer.